src/saltext/kubernetes/states/kubernetes.py

```diff
@@ -344,9 +372,80 @@ def deployment_present(
             ret["result"] = False
             ret["comment"] = str(err)
             return ret
+        ret["changes"] = {"old": deployment, "new": res}
+        ret["result"] = True
+        return ret
+
+    if deployment is not None:
+        try:
+            if source:
+                res = __salt__["kubernetes.patch_deployment"](
+                    name,
+                    namespace,
+                    patch="none",
+                    source=source,
+                    template=template,
+                    template_context=template_context,
+                    dry_run=True,
+                    **kwargs,
+                )
+            else:
+                patch_obj = {}
+                if metadata:
+                    patch_obj["metadata"] = metadata
+                if spec:
+                    patch_obj["spec"] = spec
+
+                res = __salt__["kubernetes.patch_deployment"](
+                    name,
+                    namespace,
+                    patch=patch_obj,
+                    dry_run=True,
+                    **kwargs,
+                )
+
+            if res == deployment:
```

Comment:
hint: I'd double-check that we can expect this simple check to be reliable since each dict is the result of a different API call (slight differences in return formats between API functions, missing/additional keys or time-dependent values).